src/mpi/coll/transports/generic/transport.c

```diff
@@ -0,0 +1,1243 @@
+                MPL_DBG_MSG_FMT(MPIR_DBG_COLL,VERBOSE,(MPL_DBG_FDEST,"Left reduction vtxp->nbargs.reduce_local.flags %ld \n",
+                         vtxp->nbargs.reduce_local.flags));
+                MPIR_Reduce_local((const void *) vtxp->nbargs.reduce_local.inbuf,
+                                       (void *) vtxp->nbargs.reduce_local.inoutbuf,
+                                       vtxp->nbargs.reduce_local.count,
+                                       vtxp->nbargs.reduce_local.dt, vtxp->nbargs.reduce_local.op);
+                MPL_DBG_MSG_FMT(MPIR_DBG_COLL,VERBOSE,(MPL_DBG_FDEST,"  --> GENERIC transport (reduce local_L) complete\n"));
+            } else {
+                MPL_DBG_MSG_FMT(MPIR_DBG_COLL,VERBOSE,(MPL_DBG_FDEST,"Right reduction vtxp->nbargs.reduce_local.flags %ld \n",
+                         vtxp->nbargs.reduce_local.flags));
+                /* An extra data copy is required to do the R reduction */
+                MPIR_Reduce_local((const void *) vtxp->nbargs.reduce_local.inoutbuf,
+                                       (void *) vtxp->nbargs.reduce_local.inbuf,
+                                       vtxp->nbargs.reduce_local.count,
+                                       vtxp->nbargs.reduce_local.dt, vtxp->nbargs.reduce_local.op);
+
+                MPIR_Localcopy(vtxp->nbargs.reduce_local.inbuf,
+                               vtxp->nbargs.reduce_local.count,
+                               vtxp->nbargs.reduce_local.dt,
+                               vtxp->nbargs.reduce_local.inoutbuf,
+                               vtxp->nbargs.reduce_local.count, vtxp->nbargs.reduce_local.dt);
+                MPL_DBG_MSG_FMT(MPIR_DBG_COLL,VERBOSE,(MPL_DBG_FDEST,"  --> GENERIC transport (reduce local_R) complete\n"));
+            }
+
+            /* record vertex completion */
+            MPIR_COLL_GENERIC_record_vtx_completion(vtxp, sched);
+            break;
+        }
+    }
+
+    MPIR_FUNC_VERBOSE_EXIT(MPID_STATE_MPIR_COLL_GENERIC_ISSUE_VTX);
+}
+
+#undef FUNCNAME
+#define FUNCNAME MPIR_COLL_GENERIC_poke
+/* Transport function to make progress on the schedule */
+int MPIR_COLL_GENERIC_poke(MPIR_COLL_GENERIC_sched_t * sched)
+{
+    MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_MPIR_COLL_GENERIC_POKE);
+    MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIR_COLL_GENERIC_POKE);
+
+    MPIR_COLL_GENERIC_vtx_t *vtxp;
+    int i;
+
+    /* If issued list is empty, go over the vertices and issue ready vertices
+     * Issued list should be empty only when the MPIR_COLL_GENERIC_poke function
+     * is called for the first time on the schedule */
+    if (sched->issued_head == NULL) {
+        MPL_DBG_MSG_FMT(MPIR_DBG_COLL,VERBOSE,(MPL_DBG_FDEST,"issued list is empty, issue ready vtcs\n"));
+        if (sched->total > 0 && sched->num_completed != sched->total) {
+            /* Go over all the vertices and issue ready vertices */
+            for (i = 0; i < sched->total; i++)
```

Comment:
In a crowded code block like this, there either needs to be whitespace or braces (I prefer braces).